ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {ReactiveFormsModule} from '@angular/forms';
+import {ActivatedRoute} from '@angular/router';
+import {ClarityModule} from '@clr/angular';
+import {Observable} from 'rxjs/Observable';
+
+import {ReviewStateService} from '../review-state.service';
+import {SetAnnotationItemComponent} from './set-annotation-item.component';
+
+import {
+  queryByCss,
+  updateAndTick,
+} from 'testing/test-helpers';
+
+import {
+  AnnotationType,
+  CohortAnnotationDefinition,
+  CohortAnnotationDefinitionService,
+} from 'generated';
+
+class StubRoute {
+  snapshot = {params: {
+    ns: 'workspaceNamespace',
+    wsid: 'workspaceId',
+    cid: 1
+  }};
+}
+
+class ApiSpy {
+  updateCohortAnnotationDefinition = jasmine.createSpy('updateCohortAnnotationDefinition');
+  deleteCohortAnnotationDefinition = jasmine.createSpy('deleteCohortAnnotationDefinition');
+  getCohortAnnotationDefinitions = jasmine.createSpy('getCohortAnnotationDefinitions');
+}
+
+describe('SetAnnotationItemComponent', () => {
+  let fixture: ComponentFixture<SetAnnotationItemComponent>;
+  let component: SetAnnotationItemComponent;
+
+  beforeEach(fakeAsync(() => {
+    TestBed
+      .configureTestingModule({
+        declarations: [
+          SetAnnotationItemComponent,
+        ],
+        imports: [
+          ClarityModule,
+          ReactiveFormsModule,
+        ],
+        providers: [
+          ReviewStateService,
+          {provide: CohortAnnotationDefinitionService, useValue: new ApiSpy()},
+          {provide: ActivatedRoute, useClass: StubRoute},
+        ],
+      }).compileComponents().then((resp) => {
+        fixture = TestBed.createComponent(SetAnnotationItemComponent);
+
+        // tick();
+        component = fixture.componentInstance;
+
+        // Default Inputs for tests
+        component.definition = <CohortAnnotationDefinition>{
+          cohortAnnotationDefinitionId: 1,
+          cohortId: 1,
+          columnName: 'Test Defn',
+          annotationType: AnnotationType.STRING,
+        };
+        updateAndTick(fixture);
+      });
+    // tick();
+  }));
+
+  it('Should render', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('Should set the form focus when editing', fakeAsync(() => {
+    component.editing = true;
+    updateAndTick(fixture);
+    component.setFocus();
+    // component.edit() calls setTimeout so there's still a timer on the stack
```

Comment:
Is the comment outdated? I don't follow where `component.edit()` is being called

Reply:
The comment was indeed outdated; the contents of `setFocus` used to be in `edit`